File: pi.c
#include <stdio.h>

double piFracMethod(double count, double currentPI, double k)
{
	while (count > 0)
	{
		currentPI = currentPI - (4 / (k + 2)) + (4 / (k + 4));
		k+=4;
		count--;
	}
	return currentPI;
}

int main()
{
	printf("%.28f\n",piFracMethod(500000000000,4,1));
	return 0;
}

File: pi.hs
module Pi where

piFracMethod :: Int -> Double
piFracMethod count =
	piFracHelper count 4 1
	where
		piFracHelper 0 currentPI k = currentPI
		piFracHelper count currentPI k =
			piFracHelper (count - 1) nextPI (k + 4)
			where
				nextPI = (currentPI -	(4 / (k + 2)) + (4 / (k + 4)))
	

